src/init/replace_vulnerability_detector.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+
+# Copyright (C) 2015, Wazuh Inc.
+# All rights reserved.
+# Wazuh.com
+
+# This program is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+# Shell script that updates 'Vulnerability Detector' to 'Vulnerability Detection'
+
+# Function: isConfigPresent
+# Description: Function that checks if a 'configuration' is present in the 'ossec.conf' file.
+# Parameters:
+#   $1: Path to the ossec.conf file.
+#   $2: Configuration tag.
+isConfigPresent()
+{
+    local OSSEC_CONF_PATH="$1"
+    local CONFIG_PATTERN="$2"
+
+    if ( grep -q "$CONFIG_PATTERN" "$OSSEC_CONF_PATH" ); then
+        return 0
+    fi
+
+    return 1
+}
+
+# Function: addWodleConfig
+# Description: Function that adds a wodle config in the 'ossec.conf' file.
+# Parameters:
+#   $1: Path to the ossec.conf file.
+#   $2: Path to the wodle template file.
+addWodleConfig()
+{
+    local OSSEC_CONF_PATH="$1"
+    local WODLE_TEMPLATE_FILE="$2"
+    local OSSEC_CONF_FILE_TMP="$1.tmp"
+
+    touch $OSSEC_CONF_FILE_TMP
+    cat ${OSSEC_CONF_PATH} >> $OSSEC_CONF_FILE_TMP
+
+    # Remove trailing newlines and config closure ("</ossec_config>")
+    printf "%s" "$(< ${OSSEC_CONF_FILE_TMP})" >> $OSSEC_CONF_FILE_TMP
+    sed -i '$ d' $OSSEC_CONF_FILE_TMP
+
+    # Append wodle config.
+    cat ${WODLE_TEMPLATE_FILE} >> $OSSEC_CONF_FILE_TMP
+    printf "\n" >> $OSSEC_CONF_FILE_TMP
+
+    # Close config.
+    printf "</ossec_config>\n" >> $OSSEC_CONF_FILE_TMP
+
+    mv $OSSEC_CONF_FILE_TMP $OSSEC_CONF_PATH
+}
+
+# Function: updateVulnerabilityDetector
+# Description: Function that updates the old (< v4.8) 'Vulnerability Detector' configuration with the latest one.
```

Comment:
Docu: this function is also adding the two new config blocks if they are missing. I think we should mention that in the description